api/src/main/java/org/openmrs/module/aijar/tasks/MigrateARTPatientTransfersTask.java

```diff
@@ -0,0 +1,173 @@
+import org.apache.commons.logging.LogFactory;
+import org.hibernate.Query;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.openmrs.*;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.aijar.metadata.core.EncounterTypes;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+import org.openmrs.scheduler.tasks.AbstractTask;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+import static org.openmrs.module.aijar.AijarConstants.*;
+
+public class MigrateARTPatientTransfersTask extends AbstractTask {
+    protected final Log log = LogFactory.getLog(this.getClass());
+
+    public void execute() {
+        log.info("Start Migrate Transfer Outs From Art Summary Page To Transfer Out Encounters");
+        transferOut();
+        log.info("Completed Transfer Out Migration");
+
+        log.info("Start Migrate Transfer Ins From Art Summary Page To Transfer Out Encounters");
+        transferIn();
+        log.info("Completed Transfer In Migration");
+        stopTransferService();
+    }
+
+
+    private void transferOut() {
+        String queryStringTransferOut = "from Obs obs where obs.concept =" + TRANSFER_OUT_DATE_CONCEPT_ID + " and voided=false";
+        Query query = getSession().createQuery(queryStringTransferOut);
+        List<Obs> obsList = query.list();
+        List<Encounter> encounters = new ArrayList<>();
+        List<Encounter> transferEncounters = new ArrayList<>();
+        for (Obs obs : obsList) {
+            if (!encounters.contains(obs.getEncounter())) {
+                Encounter encounter = generateTransferInOutEncounter(obs, TRANSFER_OUT_PLACE_CONCEPT_ID, EncounterTypes.TRANSFER_OUT.name(), obs.getValueDate());
+            }
+        }
+    }
+
+    private void transferIn() {
+        String queryStringTransferOut = "from Obs obs where obs.concept =" + TRANSFER_IN_CONCEPT_ID + " and voided=false";
+        Query query = getSession().createQuery(queryStringTransferOut);
+        List<Obs> obsList = query.list();
+        List<Encounter> encounters = new ArrayList<>();
+        List<Encounter> transferEncounters = new ArrayList<>();
+        for (Obs obs : obsList) {
+            if (!encounters.contains(obs.getEncounter())) {
+                Encounter encounter = generateTransferInOutEncounter(obs, TRANSFER_IN_FROM_PLACE_CONCEPT_ID, EncounterTypes.TRANSFER_IN.name(), obs.getEncounter().getEncounterDatetime());
+                ;
+            }
+        }
+    }
+
+    private void stopTransferService() {
+        TaskDefinition task = new TaskDefinition();
+        task = Context.getSchedulerService().getTaskByName(ART_TRANSFER_SCHEDULE_NAME);
+        if (task != null) {
+            try {
+                Context.getSchedulerService().shutdownTask(task);
+            } catch (SchedulerException e) {
+                e.printStackTrace();
```

Comment:
@slubwama provide an error message description of what has failed in addition to the stack trace